packages/sdk-v2/src/sdk/goodid/hooks/useCertificates.ts

```diff
@@ -1,28 +1,71 @@
-import { useCallback } from "react";
-import { isEmpty } from "lodash";
-import { CredentialTypes } from "../types";
+import { useCallback, useContext, useEffect, useState } from "react";
+import { first } from "lodash";
 
 import type { Certificate } from "../types";
 
-// This expects a single issuer (gd) for the mvp purpose
-export const useCertificates = (db: any) => {
-  const getCertificates: (credentialTypes?: string[]) => Promise<Certificate[] | undefined> = useCallback(
-    async credentialTypes => {
-      const keys = credentialTypes ?? Object.keys(CredentialTypes);
-      return await db.certificates.where("type").anyOf(keys).toArray();
+import { GoodIdContext } from "../../../contexts";
+
+const cleanupCertificate = (value: any) => {
+  if (!value) return [];
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { primary_idx, type_subject_idx, ...certificate } = value;
+  return certificate;
+};
+
+const queryCertificates = async (db: any, account: string | undefined, types: string[] | null = null) => {
+  if (!db || !account) {
+    throw new Error("Certificates database not initialized");
+  }
+
+  const accountFilter = `did:ethr:${account}`;
+  const queryField = types ? "type_subject_idx" : "credentialSubject.id";
+  const queryValue = types ? types.map(type => `${accountFilter}_${type}`) : accountFilter;
+  const queryOperator = types ? "anyOfIgnoreCase" : "equalsIgnoreCase";
+  return db.certificates.where(queryField)[queryOperator](queryValue).reverse().sortBy("issuanceDate");
+};
+
+// set types to `false` explicitly to avoid loading on mount
+// passign null or empty value will just load all certs without type filtering
+export const useCertificates = (account: string | undefined, types: string[] | null | false = null) => {
+  const [certificates, setCertificates] = useState<Certificate[]>();
+  const { db } = useContext(GoodIdContext);
+
+  const getCertificate = useCallback(
+    async (type: string[]) =>
+      queryCertificates(db, account, type)
+        .then(first) // from lodash
+        .then(cleanupCertificate),
+    [account, db]
+  );
+
+  const storeCertificate = useCallback(
+    async certificate => {
+      const { type, credentialSubject } = certificate;
+      const typeSubjectIdx = `${credentialSubject.id}_${type[1]}`;
```

Comment:
@johnsmith-gooddollar mapping here on types create unneccessary duplicate s of the same credential

Reply:
REVERT! it's multi entry index **required** to properly filter by type! otherwise only one type will be indexed per each certificate !!!!

Reply:
https://dexie.org/docs/MultiEntry-Index see, the field value **should be** array